### DISCUSSION TITLE
Brand based category images

### DISCUSSION THREAD

**Discussion by inalto** [OFFLINE-GmbH/oc-mall-plugin#807] [Ideas]
I am planning to extend the category model to allow to have different category images for different brands.
The idea is to have a jsonable repeating field that binds an image choosen via the media manager to a brand.
Using of jsonable should not impact in performance.
This open the possibility to create component blocks that link to a category, which is handy in sites with multiple brands like the one i am working on.
And of course if a brand category image is missing we can fall back to the default category image.
I also imagined to build an external plugin but this is less performant and has a less simple UI 

Let me know if someone is interested or has different ideas so i can create a pull request.

**Comment by tobias-kuendig** [comment]
To me this sounds like a rather specific use-case that probably is not a good fit for the default plugin feature set. 

You should be able to come up with a solution using [October's plugin extension API](https://octobercms.com/docs/plugin/extending). If you need any more events to integrate your changes, feel free to suggest them here. 

Also, if you have a working solution, don't forget to share it!
